content/en/docs/Getting started/configuration.md

```diff
@@ -0,0 +1,227 @@
+cat: /etc/hostname: No such file or directory
+```
+
+## Configuring Kairos
+
+So how do you configure Kairos if it is immutable? The answer is simple: you use configuration files. These files are read at boot time and the changes are applied to the OS on every boot.
+
+Your system is already configured this way. Have a look at the file `/oem/90_custom.yaml`. If you followed the [Getting Started guide](/docs/getting-started), you should see a file with similar content to mine:
+
+```yaml
+#cloud-config
+
+install:
+    device: /dev/vda
+k3s:
+    enabled: true
+name: Config generated by the installer
+stages:
+    network:
+        - users:
+            kairos:
+                groups:
+                    - admin
+                name: kairos
+                passwd: kairos
+                ssh_authorized_keys:
+                    - github:mauromorales
+```
+
+The name of the file is not important, but the extension and the location are. You must add your configuration files to `/oem/` and they must have a `.yaml` extension. Finally, the first line of the file must be `#cloud-config` to be recognized by the system.
+
+The `install` directive is used to give instructions on how to install the system when it runs on the first boot, so let's ignore it for now.
+
+The `k3s` directive is used to enable or disable the Kubernetes distribution that comes with Kairos. If you set it to `false`, the K3s services will not start on boot.
+
+The `name` directive can be added at every level of the configuration file. It is useful to distinguish between different steps.
+
+The `stages` directive is used to define the different stages of the configuration. The `network` stage is used to configure the system when network starts. In this case, we create a user called `kairos` with the password `kairos` and an SSH key.
+
+## Configuring the hostname
+
+Configuring the hostname is a simple task. All we have to do is add the `hostname` directive at the root level of the configuration file with the value we want to give to this node. The order of the directives isot important. Let's add it after the `k3s` directive:
+
+```yaml
+#cloud-config
+
+install:
+    device: /dev/vda
+k3s:
+    enabled: true
+hostname: "master-node"
+name: Config generated by the installer
+stages:
+    network:
+        - users:
+            kairos:
+                groups:
+                    - admin
+                name: kairos
+                passwd: kairos
+                ssh_authorized_keys:
+                    - github:mauromorales
+```
+
+Save the file and reboot the node. After the reboot, you should see the new hostname on the prompt.
+
+```bash
+kairos@master-node:~$
+```
+
+Let's take this a bit further. In the future we want to add a second master node to our cluster. Let's then use a value that is unique to this node and add it as a suffix.
+
+```yaml
+#cloud-config
+
+install:
+    device: /dev/vda
+k3s:
+    enabled: true
+hostname: "master-{{ trunc 4 .MachineID }}"
```

Comment:
Maybe we should point the users to some docs on what templating functions are available? To at least explain why this works.

Reply:
@jimmykarily very good point, I was looking for it on the docs but couldn't find a place where we define this, do you know where they come from?

Reply:
found it http://masterminds.github.io/sprig/

Reply:
I wanna see you documenting this part :D https://github.com/mudler/yip/blob/3eeeb15bc6b921a977c1f44cf30e77c81dcd4dc6/pkg/plugins/hostname.go#L42

Reply:
I think user also gets access to these: https://github.com/zcalusic/sysinfo
because of this: https://github.com/mudler/yip/blob/3eeeb15bc6b921a977c1f44cf30e77c81dcd4dc6/pkg/plugins/common.go#L47

but I'm not 100% sure